### ISSUE TITLE
jQuery parser

### ISSUE THREAD

**Issue by jmrocela** [jmrocela/munchjs#21]

**Comment by rafaelnco** [comment]
Is this about all setters without parameters in jQuery`s (1.8.3) jsdom (0.6.5) lib level1/core.js and level2/html.js? Really need some support with this...

**Comment by arlusf** [comment]
I'm not sure what your trouble is? I'm not the one to know everything under
the sun, or maybe I am, but surely there is a solution that I can help out
with and let the community know if this might help them also.

**Comment by rafaelnco** [comment]
_note: I'm using a the original package.json:_

```
package.json [51]:
    "jquery": "~1.8.3",
    "jsdom": "~0.6.5",
```

**TL;DR:
My problem is with JSDom 0.6.5 (used as a jQuery lib). It has many setter functions without parameters and it seems to be a problem related to jQuery 1.8.3.

I've modified the source code (adding the parameters to each function) but I'm not sure my 'solution' (if I can say that) is the right one. That's why I'm asking for your support as I'm using the same jQuery and JSDom versions found in the original package.json.
**

@arlusf I've registered an [issue about this](https://github.com/jmrocela/munchjs/issues/25) in which you have commented and I really appreciate your support! The only thing I'm afraid of is that your comments [actually relate to this issue](https://github.com/jmrocela/munchjs/issues/26). Anyways, there is a better explanation for this:

**the code below was extracted from** _/usr/lib/node_modules/munch/node_modules/jquery/node_modules/jsdom/lib/jsdom/level1/core.js_

```
level1/core.js [418]:
...
set nodeName() { throw new core.DOMException();},
  get attributes() { return this._attributes;},
  get firstChild() {
    return this._childNodes.length > 0 ? this._childNodes[0] : null;
  },
  set firstChild() { throw new core.DOMException();},
  get ownerDocument() { return this._ownerDocument;},
  get readonly() { return this._readonly;},

  get lastChild() {
    var len = this._childNodes.length;
    return len > 0 ? this._childNodes[len -1] : null;
  },
  set lastChild() { throw new core.DOMException();},
... 
```

There are currently 13 setters which triggers sintax errors when MunchJs is executed:

```
/usr/lib/node_modules/munch/node_modules/jquery/node_modules/jsdom/lib/jsdom/level1/core.js:418
set nodeName() { throw new core.DOMException();},
              ^^

SyntaxError: Setter must have exactly one formal parameter.
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/usr/lib/node_modules/munch/node_modules/jquery/node_modules/jsdom/lib/jsdom/level2/core.js:1:94)
    at Module._compile (module.js:397:26)
    at Object.Module._extensions..js (module.js:404:10)
```

I was struggling with this until I finally  modified the code as below:

```
level1/core.js [418]:
...
  set nodeName(a) { throw new core.DOMException();},
  get attributes() { return this._attributes;},
  get firstChild() {
    return this._childNodes.length > 0 ? this._childNodes[0] : null;
  },
  set firstChild(a) { throw new core.DOMException();},
  get ownerDocument() { return this._ownerDocument;},
  get readonly() { return this._readonly;},

  get lastChild() {
    var len = this._childNodes.length;
    return len > 0 ? this._childNodes[len -1] : null;
  },
  set lastChild(a) { throw new core.DOMException();},
...
```

So, running MunchJS with the modified code led me to the error solved by [this pull request](https://github.com/jmrocela/munchjs/pull/24) (which I've added to my local munch.js). Now, my 'struggle' relates to [this issue](https://github.com/jmrocela/munchjs/issues/26).

**Comment by rafaelnco** [comment]
# You can reproduce this error by executing a fresh munchjs install!
